### DISCUSSION TITLE
Problems with cmdVel(roll, pitch, yawrate, thrust).

### DISCUSSION THREAD

**Discussion by Wallat1** [USC-ACTLab/crazyswarm#683] [Q&A]
Hi.

I have used the commands takeoff and land and I have not had problems, but when using the cmdVel(0, 20, 0, 90), the crazyflie make nothing, for this reason I would like to know if is it necessary to make use of  other commands? or if do you have an example of how to use it?.

I would appreciate your help.

Regards.

**Comment by whoenig** [comment]
The firmware has a safeguard that first requires you to send (0,0,0,0) a few times. Other than that, you can simply use low- and high-level commands. Just switching from low- back to high-level is a bit more tricky.

**Reply by Wallat1** [reply]
Hi.

Thank you for replying, now my question is whether even when I use the takeoff command is it necessary to use cmdVel(0, 0, 0, 0)?.

Regards.

**Reply by whoenig** [reply]
Not needed in that case.

**Reply by Wallat1** [reply]
Hi. 

I have tried to replicate the program ramp.py of crazyflie-lib-python, but it does not work, I attach code
![20220919_154351](https://user-images.githubusercontent.com/112857979/191158513-cc3ce627-8ad4-4e0d-ad46-5bf9afa37a28.jpg)

I do not know that it could be wrong with the code.

I would appreciate your help.

Regards.

**Reply by whoenig** [reply]
Could you specify in which way it doesn't work?

**Reply by Wallat1** [reply]
Hi.

When I run the program ramp.py the motors turn on, but in this case the crazyflie does nothing, just the led blinks when executing the program.  

I hope you can help me.

Regards.

**Reply by whoenig** [reply]
Your code looks fine. Is the CF connected properly? Are there any relevant logging messages?

**Comment by Wallat1** [comment]
Hi. 

Yes, the CF is connected properly and I do not get any message, for that I do not understand what can be wrong with  the code.

Regards.

**Comment by knmcguire** [comment]
Do you have a controller by any chance? Technically the standard launch file [hover_swarm.launch](https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/launch/hover_swarm.launch) enables connecting an controller and that is able to use the same cmd_vel as well. 

Also could you try to add a sleep to the while loop too?

**Reply by Wallat1** [reply]
Hi.

Yes, I have a ps4 controller, but I don't know what file I have to modify to use cmdVel() , could you tell me what I should modify please?. On the other hand  I know that the controller is working because when I press the R2 button the CF takes off and with L2 it lands.

Regards.
